src/main/java/games/strategy/util/Util.java

```diff
@@ -122,4 +130,19 @@ public static String getStringFromInputStream(InputStream in) {
     }
     return builder.toString();
   }
+
+  /**
+   * Creates a hash of the given String based on the SHA-512 algorithm.
+   * 
+   * @param input The input String to hash.
+   * @return A hashed hexadecimal String of the input.
+   */
+  public static String sha512(String input) {
```

Comment:
I only found this in one place in the production code. I would recommend having it as a package protected method in that class and mark it as `@VisibleForTesting`. That way it is more clear the usage is constrained to the one location. Otherwise, if there are multiple locations that use it, I think that indicates the abstraction we smell by having such a utility method. There are some chunks of code here and there that do encryption and use sha512, perhaps those should be grouped to their own class. the goal would be to move this method around the things that use it, so it would all be in the same class and `sha512` could be private.